controls/radrichtextbox/getting-started/formatting-via-a-sample-ui.md

```diff
@@ -0,0 +1,148 @@
+        this.radRichTextBox.Focus(); //return focus to the control
+    }
+{{endregion}}
+
+
+
+#### __[VB.NET] Example 2: Using the Formatting API in Code-Behind__
+
+{{region radrichtextbox-getting-started-formatting-via-a-sample-ui_2}}
+	Private Sub BoldButton_Click(sender As Object, e As RoutedEventArgs)
+	    Me.radRichTextBox.ToggleBold()
+	    Me.radRichTextBox.Focus()
+	End Sub
+	Private Sub ItalicButton_Click(sender As Object, e As RoutedEventArgs)
+	    Me.radRichTextBox.ToggleItalic()
+	    Me.radRichTextBox.Focus()
+	End Sub
+	Private Sub UnderlineButton_Click(sender As Object, e As RoutedEventArgs)
+	    Me.radRichTextBox.ToggleUnderline()
+	    Me.radRichTextBox.Focus()
+	End Sub
+{{endregion}}
+
+
+
+The UI should also respond when the caret is on a document position where the text is modified. For example, the __BoldButton__ should be toggled if the caret is on bold text. This can be done by handling the __ToggleStateChanged__ event as demonstrated in **Example 3**. 
+        
+
+#### __[C#] Example 3: Changing the Toggle State of a Button Depending on the Caret Position__
+
+{{region radrichtextbox-getting-started-formatting-via-a-sample-ui_3}}
+        InitializeComponent();
+
+        this.radRichTextBox.Commands.ToggleBoldCommand.ToggleStateChanged += new EventHandler<Telerik.Windows.Documents.RichTextBoxCommands.StylePropertyChangedEventArgs<bool>>(ToggleBoldCommand_ToggleStateChanged);
+        this.radRichTextBox.Commands.ToggleItalicCommand.ToggleStateChanged += new EventHandler<Telerik.Windows.Documents.RichTextBoxCommands.StylePropertyChangedEventArgs<bool>>(ToggleItalicCommand_ToggleStateChanged);
+        this.radRichTextBox.Commands.ToggleUnderlineCommand.ToggleStateChanged += new EventHandler<Telerik.Windows.Documents.RichTextBoxCommands.StylePropertyChangedEventArgs<bool>>(ToggleUnderlineCommand_ToggleStateChanged);
+    }
+
+    void ToggleBoldCommand_ToggleStateChanged(object sender, Telerik.Windows.Documents.RichTextBoxCommands.StylePropertyChangedEventArgs<bool> e)
+    {
+        BoldButton.IsChecked = e.NewValue;
+    }
+
+    void ToggleUnderlineCommand_ToggleStateChanged(object sender, Telerik.Windows.Documents.RichTextBoxCommands.StylePropertyChangedEventArgs<bool> e)
+    {
+        UnderlineButton.IsChecked = e.NewValue;
+    }
+
+    void ToggleItalicCommand_ToggleStateChanged(object sender, Telerik.Windows.Documents.RichTextBoxCommands.StylePropertyChangedEventArgs<bool> e)
+    {
+        ItalicButton.IsChecked = e.NewValue;
+    }
+{{endregion}}
+
+
+
+#### __[VB.NET] Example 3: Changing the Toggle State of a Button Depending on the Caret Position__
+
+{{region radrichtextbox-getting-started-formatting-via-a-sample-ui_4}}
+    Public Sub New()
+        InitializeComponent()
+
+        Me.radRichTextBox.Commands.ToggleBoldCommand.ToggleStateChanged += New EventHandler(Of Telerik.Windows.Documents.RichTextBoxCommands.StylePropertyChangedEventArgs(Of Boolean))(ToggleBoldCommand_ToggleStateChanged)
```

Comment:
Isn't it possible for all buttons to be attached to the same handler and to use the sender instead of the button's name?